keras_nlp/metrics/edit_distance.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Edit Distance metric implementation based on `keras.metrics.Metric`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class EditDistance(keras.metrics.Metric):
+    """Edit Distance.
+
+    This class implements the edit distance metric, sometimes called
```

Comment:
This class implements edit distance, sometime called Levenshtein Distance, as a `tf.keras.Metric`.

backticked symbol names will get autolinked in our keras.io docs. So listing key symbols will help users navigate.